Tests/ReactiveSwiftTests/SignalSpec.swift

```diff
@@ -200,6 +200,51 @@ class SignalSpec: QuickSpec {
 			}
 		}
 
+		describe("reentrant") {
+			#if arch(x86_64) && canImport(Darwin)
```

Comment:
What's the reason for restricting this text to x86?

Reply:
Catching deadlock by intercepting signals is supported only in iOS simulators.

> https://github.com/mattgallagher/CwlPreconditionTesting#cwlpreconditiontesting
>
> A Mach exception handler, written in Swift and Objective-C, that allows EXC_BAD_INSTRUCTION (as raised by Swift's assertionFailure/preconditionFailure/fatalError) to be caught and tested.
>
> NOTE: the iOS code runs in the simulator only. It is for logic testing and cannot be deployed to the device due to the Mach exception API being private on iOS.

Reply:
What about Macs with an M1?

Reply:
It is more a question for Nimble/CwlPreconditionSignal. If they support catching signals on M1, we can lift this.